### PR TITLE
docs: add contributing guides

### LINKED ISSUE #234: Add contributions guidelines
Let's define how we want to receive community contributions, so that it's even much easier to do it.

Possibly a CONTRIBUTING.md file will be enough, although a good PULL_REQUEST_TEMPLATE can help

### PR DESCRIPTION
**Pull request by mdelapenya**
## What does this PR do?
It adapts the contributing guides from tc-java to tc-go, removing what it's not needed.

## Why is it important?
The community should have a clear understanding of what are the expectations of the maintainers when receiving a contribution.

This document is a live one, which means that we will be adding and/or removing sections if/when needed.

## Related issues
- Closes #234

### CODE REVIEW

File: docs/contributing.md [hunk 1]
```diff
@@ -1,4 +1,25 @@
...
+    * follow the style, naming and structure conventions of the rest of the project.
+    * make commits atomic and easy to merge.
+    * when updating documentation, please see [our guidance for documentation contributions](contributing_docs.md).
+    * apply format running `go fmt`, or the shell script `./scripts/checks.sh`
+    * verify all tests are passing. Build and test the project with `make test-all` to do this.
+
+## Combining Dependabot PRs
+
+Since we generally get quite a few Dependabot PRs, we regularly combine them into single commits. 
+For this, we are using the [gh-combine-prs](https://github.com/rnorth/gh-combine-prs) extension for [GitHub CLI](https://cli.github.com/).
```
Comment (kiview) [inline docs/contributing.md:17]: Did you ever use the GH CLI plugin before in `tc-go`?

**Review by kiview — APPROVED** [review]

File: docs/contributing.md (hunk 1 above)
Comment (mdelapenya) [reply docs/contributing.md:17]: No, and I was tempted to remove the entire section, as I copied it from tc-java, and we do not receive as much @dependabot PRs (https://github.com/testcontainers/testcontainers-go/pulls?q=is%3Apr+author%3Aapp%2Fdependabot+is%3Aclosed)

In fact, I think we should remove it

Comment (kiview) [reply docs/contributing.md:17]: Let's remove it if this does not reflect the reality 👍

Comment (mdelapenya) [reply docs/contributing.md]: Done, thanks!

**Review by eddumelendez — APPROVED** [review]

**Pull request merged**